packages/editor-ui/src/utils/__tests__/nodeTypesUtils.spec.ts

```diff
@@ -0,0 +1,283 @@
+						{
+							id: '135e0eb0-f412-430d-8990-731c57cf43ae',
+							name: 'document',
+							value: "={{ $('Edit Fields 2').item.json.document}}",
+							type: 'string',
+						},
+					],
+				},
+				includeOtherFields: false,
+				options: {},
+			},
+			type: 'n8n-nodes-base.set',
+			typeVersion: 3.4,
+			position: [560, -140],
+			id: '7306745f-ba8c-451d-ae1a-c627f60fbdd3',
+			name: 'Edit Fields 2',
+		},
+		expected: ['Edit Fields 2'],
+	},
+	{
+		caseName: 'Should handle expressions with single and double quotes',
+		node: {
+			parameters: {
+				authentication: 'oAuth2',
+				resource: 'sheet',
+				operation: 'read',
+				documentId: {
+					__rl: true,
+					value: "={{ $('Edit Fields').item.json.document }}",
+					mode: 'id',
+				},
+				sheetName: {
+					__rl: true,
+					value: '={{ $("Edit Fields 2").item.json.sheet }}',
+					mode: 'id',
+				},
+				filtersUI: {},
+				combineFilters: 'AND',
+				options: {},
+			},
+			type: 'n8n-nodes-base.googleSheets',
+			typeVersion: 4.5,
+			position: [440, 0],
+			id: '9a95ad27-06cf-4076-af6b-52846a109a8b',
+			name: 'Google Sheets',
+			credentials: {
+				googleSheetsOAuth2Api: {
+					id: '8QEpi028oHDLXntS',
+					name: 'milorad@n8n.io',
+				},
+			},
+		},
+		expected: ['Edit Fields', 'Edit Fields 2'],
+	},
+	{
+		caseName: 'Should only add one reference for each referenced node',
+		node: {
+			parameters: {
+				authentication: 'oAuth2',
+				resource: 'sheet',
+				operation: 'read',
+				documentId: {
+					__rl: true,
+					value: "={{ $('Edit Fields').item.json.document }}",
+					mode: 'id',
+				},
+				sheetName: {
+					__rl: true,
+					value: "={{ $('Edit Fields').item.json.sheet }}",
+					mode: 'id',
+				},
+				filtersUI: {},
+				combineFilters: 'AND',
+				options: {},
+			},
+			type: 'n8n-nodes-base.googleSheets',
+			typeVersion: 4.5,
+			position: [440, 0],
+			id: '9a95ad27-06cf-4076-af6b-52846a109a8b',
+			name: 'Google Sheets',
+			credentials: {
+				googleSheetsOAuth2Api: {
+					id: '8QEpi028oHDLXntS',
+					name: 'milorad@n8n.io',
+				},
+			},
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should handle multiple node references in one expression',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('Edit Fields').item.json.one }} {{ $('Edit Fields 2').item.json.two }} {{ $('Edit Fields').item.json.three }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields', 'Edit Fields 2'],
+	},
+	{
+		caseName: 'Should ignore spaces around node references',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('   Edit Fields     ').item.json.one }}",
```

Comment:
same as below. this case does not make sense. this should not evaluate in n8n even if node existed without the spaces.

Reply:
This should evaluate to `   Edit Fields     `